unstructured/chunking/title.py

```diff
@@ -9,105 +10,153 @@
     Element,
     ElementMetadata,
     Table,
+    TableChunk,
     Text,
     Title,
 )
 
 
+def chunk_table_element(
+    element: Table,
+    max_characters: Optional[int] = 1500,
+) -> List[Union[Table, TableChunk]]:
+    chunks = []
+
+    element_char_len = len(element.text)
+
+    html_table = element.text_as_html if hasattr(element, "text_as_html") else None
+    if html_table:
+        element_char_len = len(html_table)
+    if element_char_len <= max_characters:  # type: ignore
+        chunks.append(element)
+    else:
+        text = element.text
+        text_as_html = element.text_as_html if hasattr(element, "text_as_html") else None
+        i = 0
+        metadata = element.metadata
+        while text or text_as_html:
+            text_chunk = text[:max_characters]
+            table_chunk = TableChunk(
+                text=text_chunk,
+                metadata=copy.copy(metadata),
+            )
+            if text_as_html:
+                text_as_html_chunk = text_as_html[:max_characters]
+                table_chunk.metadata.text_as_html = text_as_html_chunk
+                # Remove the processed chunk from text_as_html
+                text_as_html = text_as_html[max_characters:]
+            if i > 0:
+                table_chunk.metadata.is_continuation = True
+
+            chunks.append(table_chunk)
+            i += 1
+
+            # Remove the processed chunk from text
+            text = text[max_characters:]
+
+            # Ensure that text and text_as_html are not empty before continuing
+            if not text and not text_as_html:
+                break
+    return chunks
+
+
 def chunk_by_title(
     elements: List[Element],
     multipage_sections: bool = True,
-    combine_under_n_chars: int = 500,
-    new_after_n_chars: int = 1500,
+    combine_text_under_n_chars: int = 500,
+    max_characters: int = 1500,
 ) -> List[Element]:
     """Uses title elements to identify sections within the document for chunking. Splits
     off into a new section when a title is detected or if metadata changes, which happens
     when page numbers or sections change. Cuts off sections once they have exceeded
-    a character length of new_after_n_chars.
+    a character length of max_characters.
 
     Parameters
     ----------
     elements
-        A list of unstructured elements. Usually the ouput of a partition functions.
+        A list of unstructured elements. Usually the output of a partition functions.
     multipage_sections
         If True, sections can span multiple pages. Defaults to True.
-    combine_under_n_chars
+    combine_text_under_n_chars
         Combines elements (for example a series of titles) until a section reaches
         a length of n characters.
-    new_after_n_chars
+    max_characters
```

Comment:
`new_after_n_chars` was a "soft limit".
theoretically, soft limits are nice because one is never going to chop an element in half, for instance.

i can see wanting both the soft limit and hard limit (`max_characters`).

for example, if my accumulated chunk is over 800 chars (a soft limit), start a new clean chunk. but don't let the chunk exceed 1000 chars (a hard limit).

imo, ^^ is worth a follow on github issue.

Reply:
the problem of giant elements could exist anywhere, i.e., we should chunk a giant NarrativeText block the same way we would a Table very soon if not in this PR.

> So basically we'll have a soft and hard max but table is only using hard max and title is only using soft max until the next ticket/pr :)

i think that works, as long as we are prioritizing having a soft max and hard max that is consistent across element types soon afterwards.

Reply:
re: soon - Yes definitely! Will add this context to the tickets in the next sprint